### PR TITLE
chore(generator): Response and Request

### PR DESCRIPTION
**Pull request by avodovnik**

### CODE REVIEW

**Review by kblok — CHANGES_REQUESTED** [review]

File: src/PlaywrightSharp/Contracts/IRequest.generated.cs [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+		/// </para>
+		/// </summary>
+		RequestTimingResult Timing { get; }
+	
+		/// <summary><para>URL of the request.</para></summary>
+		string Url { get; }
+
+        /// <summary><para>Returns the <see cref="JsonDocument"/> representation of <see cref="PostDataBuffer"/>.</para></summary>
+        /// <param name="documentOptions">The JSON Document Options.</param>
+        JsonDocument GetPayloadAsJson(JsonDocumentOptions documentOptions = default);
```
[the remark this answers is not in the document]
Comment (avodovnik) [reply src/PlaywrightSharp/Contracts/IRequest.generated.cs:161]: Yes, here's the upstream PR: https://github.com/microsoft/playwright/pull/5763

File: src/PlaywrightSharp/Contracts/IResponse.generated.cs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	public partial interface IResponse
+	{
+		/// <summary><para>Returns the buffer with response body.</para></summary>
+		Task<byte[]> GetBodyAsync();
+
+        /// <summary><para>Waits for this response to finish, returns failure error if request failed.</para></summary>
+        Task<string> GetFinishedAsync();
+
+        /// <summary><para>Returns the <see cref="IFrame"/> that initiated this response.</para></summary>
+        IFrame Frame { get; }
```
[the remark this answers is not in the document]
Comment (avodovnik) [reply src/PlaywrightSharp/Contracts/IResponse.generated.cs:58]: It's not. It happened when I pasted things around. I'll fix it as I regenerate.

File: src/PlaywrightSharp/Contracts/IResponse.generated.cs [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+		bool Ok { get; }
+	
+		/// <summary><para>Returns the matching <see cref="IRequest"/> object.</para></summary>
+		IRequest Request { get; }
+	
+		/// <summary><para>Contains the status code of the response (e.g., 200 for a success).</para></summary>
+		int Status { get; }
+
+        /// <summary><para>Gets the <see cref="System.Net.HttpStatusCode"/> code of the response.</para></summary>
+        System.Net.HttpStatusCode StatusCode { get; }
```
[the remark this answers is not in the document]
Comment (avodovnik) [reply src/PlaywrightSharp/Contracts/IResponse.generated.cs:89]: It's auto-generated now :-) See same upstream PR: https://github.com/microsoft/playwright/pull/5763

File: src/PlaywrightSharp/Helpers/JsonExtensions.cs [hunk 4]
```diff
@@ -71,7 +71,6 @@ internal static JsonSerializerOptions GetNewDefaultSerializerOptions(bool ignore
                 IgnoreNullValues = ignoreNullValues,
                 Converters =
                 {
-                    new FlexibleStringEnumConverter<ResourceType>(ResourceType.Unknown),
```
[the remark this answers is not in the document]
Comment (avodovnik) [reply src/PlaywrightSharp/Helpers/JsonExtensions.cs:74]: Not for `ResourceType` though. It's removed.

File: src/PlaywrightSharp/Contracts/IRequest.generated.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (kblok) [reply src/PlaywrightSharp/Contracts/IRequest.generated.cs:161]: Can we fix the indentation :neckbeard:

File: src/PlaywrightSharp.Tests/TestConstants.cs [hunk 5]
```diff
@@ -16,7 +16,7 @@ internal static class TestConstants
         public const int DefaultTaskTimeout = 5_000;
 
         public static string Product => string.IsNullOrEmpty(Environment.GetEnvironmentVariable("PRODUCT")) ?
-            ChromiumProduct :
+            FirefoxProduct :
```
Comment (kblok) [inline src/PlaywrightSharp.Tests/TestConstants.cs:19]: Can we rollback to the default?

**Review by kblok — APPROVED** [review]

File: src/PlaywrightSharp/ResourceType.cs [hunk 6]
```diff
@@ -1,112 +0,0 @@
...
-using System.Text.Json.Serialization;
-using PlaywrightSharp.Helpers;
-
-namespace PlaywrightSharp
-{
-    /// <summary>
-    /// Resource type.
-    /// </summary>
-    /// <seealso cref="IRequest.ResourceType"/>
-    public enum ResourceType
```
[the remark this answers is not in the document]
Comment (avodovnik) [reply src/PlaywrightSharp/ResourceType.cs:11]: https://github.com/microsoft/playwright-sharp/pull/1270/files#diff-24b7973e733c698122de731faa471e33fa2bec65b4466a2d5635911a517825ef

**Review by kblok — APPROVED** [review]

**Pull request merged**